### PR TITLE
feat(robo2vlm_sft): support qwen3 vl model sft in rlinf

### PR DESCRIPTION
**Pull request by FxxxxU**
### Description

Add support for supervised fine-tuning (SFT) of the Qwen3-VL-4B model in Robo2VLM and improve the evaluation pipeline for VLM SFT. Fix the previous evaluation bug when enabling full_shard under FSDP. Add the Qwen3-VL SFT shell scripts and corresponding CI tests.

### Motivation and Context

The previous PR contained several bugs. This PR fixes those issues and, based on the earlier work, adds experimental evaluation for the Qwen3-VL model.

### How has this been tested?

This PR has been validated through experiments. The experiment scripts have been uploaded, and the results are shown below. 
I conducted SFT training for 6,000 iterations on the Qwen3-VL-4B model using the Robo2VLM-1 dataset
(https://huggingface.co/datasets/keplerccc/Robo2VLM−1) with 4 H100 Nvidia GPU. 

loss:

tTrgk606a0

grad_norm:

NyB53BKEyG

The dataset was split into training and test sets during training, and evaluation was performed on 6,676 test samples. The final evaluation accuracy reached 96.9%.

VT1DWtewNY

Additionally, new CI tests have been added.

### Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Documentation update (Document-only update)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Review by guozhen1997 — APPROVED** [review]

File: rlinf/hybrid_engines/fsdp/utils.py [hunk 1]
```diff
@@ -827,3 +829,160 @@ def unpack_fsdp_logprobs(
...
+def generate_with_kv_cache(
+    model: Union[FSDP, FSDPModule],
+    eos_token_id: int,
+    pad_token_id: int,
+    amp_context: ContextManager,
+    input_ids: torch.Tensor,
+    attention_mask: torch.Tensor,
+    multi_modal_inputs: dict[str, torch.Tensor],
+    max_new_tokens: int = 128,
+) -> torch.Tensor:
```
Comment (qurakchin) [inline rlinf/hybrid_engines/fsdp/utils.py:843]: add a comment to declare why to add this fuction

Comment (FxxxxU) [reply rlinf/hybrid_engines/fsdp/utils.py:843]: Done.
